File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        
        // 우선순위 큐를 내림차순으로 정렬 = 우선순위가 높은 것부터 뽑겠다
        PriorityQueue<Integer> q = new PriorityQueue<>(Collections.reverseOrder());
        for (int i : priorities) {
            q.add(i);
        }
        
        int cnt = 0;
        
        while(!q.isEmpty()) {
            for (int i=0; i<priorities.length; i++) {
                if (priorities[i] == q.peek()) {
                    q.poll();
                    cnt++;
                    
                    if (i == location) {
                        return cnt;
                    }
                }
            }
        }
        
        return cnt;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let q = priorities.map((priority, index) => ({index, priority}));
    
    let cnt = 0;
    
    while(q.length > 0) {
        let cur = q.shift();
        
        let hasHigherPriority = q.some(process => process.priority > cur.priority);
        
        if (hasHigherPriority) {
            q.push(cur);
        } else {
            cnt++;
            if (cur.index === location) {
                return cnt;
            }
        }
    }
}